sql/wsrep_mysqld.cc

```diff
@@ -660,15 +660,13 @@ int wsrep_init_server()
     std::string working_dir;
     wsrep::gtid initial_position;
 
-    if (WSREP_ON)
-    {
-      server_name= wsrep_server_name();
-      server_id= wsrep_server_id();
-      node_address= wsrep_server_node_address();
-      incoming_address= wsrep_server_incoming_address();
-      working_dir= wsrep_server_working_dir();
-      initial_position= wsrep_server_initial_position();
-    }
+    server_name= wsrep_server_name();
+    server_id= wsrep_server_id();
+    node_address= wsrep_server_node_address();
+    incoming_address= wsrep_server_incoming_address();
+    working_dir= wsrep_server_working_dir();
+    initial_position= wsrep_server_initial_position();
+
```

Comment:
What is the reason for this change?

Reply:
Wsrep_server_state::init_once is called from wsrep_init_server() and it will initialize singleton from there. It requires correct data to be provided.
wsrep.variables fails because it provides empty working_dir.